src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,1295 @@
+_INTEREST_FIELD: str = "interest"
+_ISDUST: str = "isDust"
+_ISFIATPAYMENT: str = "isFiatPayment"
+_LEGALMONEY: str = "legalMoney"
+_LENDINGTYPE: str = "lendingType"
+_LIMIT: str = "limit"
+_LOCKPERIOD: str = "lockPeriod"
+_OBTAINAMOUNT: str = "obtainAmount"
+_ORDER: str = "order"  # CCXT only variable
+_ORDERNO: str = "orderNo"
+_PAGEINDEX: str = "pageIndex"
+_PAGESIZE: str = "pageSize"
+_POSITIONID: str = "positionId"
+_PRICE: str = "price"
+_PRODUCT: str = "product"
+_PROFITAMOUNT: str = "profitAmount"
+_REDEMPTION: str = "REDEMPTION"
+_ROWS: str = "rows"
+_SELL: str = "sell"  # CCXT only variable
+_SIDE: str = "side"  # CCXT only variable
+_SIZE: str = "size"
+_STAKING: str = "STAKING"
+_STARTTIME: str = "startTime"
+_STATUS: str = "status"
+_SOURCEAMOUNT: str = "sourceAmount"
+_SUBSCRIPTION: str = "SUBSCRIPTION"
+_SYMBOL: str = "symbol"
+_TIME: str = "time"
+_TIMESTAMP: str = "timestamp"  # CCXT only variable
+_TRANID: str = "tranId"
+_TRANSACTIONTYPE: str = "transactionType"
+_TOTAL: str = "total"
+_TOTALFEE: str = "totalFee"
+_TOTALNUM: str = "totalNum"
+_TYPE: str = "type"
+_TXID: str = "txid"  # CCXT doesn't capitalize I
+_TXNTYPE: str = "txnType"
+_UPDATETIME: str = "updateTime"
+_USERNAME: str = "userName"
+_WITHDRAWAL: str = "withdrawal"  # CCXT only variable
+
+# Time period constants
+_NINETY_DAYS_IN_MS: int = 7776000000
+_THIRTY_DAYS_IN_MS: int = 2592000000
+_ONE_DAY_IN_MS: int = 86400000
+_MS_IN_SECOND: int = 1000
+
+# Record limits
+_DEPOSIT_RECORD_LIMIT: int = 1000
+_DIVIDEND_RECORD_LIMIT: int = 500
+_DUST_TRADE_RECORD_LIMIT: int = 100
+_INTEREST_SIZE_LIMIT: int = 100
+_MINING_PAGE_LIMIT: int = 200
+_TRADE_RECORD_LIMIT: int = 1000
+_WITHDRAWAL_RECORD_LIMIT: int = 1000
+
+# Types of Binance Dividends
+_BNBVAULT = "BNB Vault"
+_ETHSTAKING = "ETH 2.0 Staking"
+_FLEXIBLESAVINGS = "Flexible Savings"
+_LAUNCHPOOL = "Launchpool"
+_LOCKEDSAVINGS = "Locked Savings"
+_LOCKEDSTAKING = "Locked Staking"
+_SOLOAIRDROP = "SOLO airdrop"
+
+_AIRDROP_LIST = [_SOLOAIRDROP]
+_INTEREST_LIST = [_FLEXIBLESAVINGS, _LOCKEDSAVINGS]
+_STAKING_LIST = [_ETHSTAKING, _LOCKEDSTAKING, _BNBVAULT, _LAUNCHPOOL]
+
+
+class _ProcessAccountResult(NamedTuple):
+    in_transactions: List[InTransaction]
+    out_transactions: List[OutTransaction]
+    intra_transactions: List[IntraTransaction]
+
+
+class _Trade(NamedTuple):
+    base_asset: str
+    quote_asset: str
+    base_info: str
+    quote_info: str
+
+
+class InputPlugin(AbstractInputPlugin):
+
+    __BINANCE_COM: str = "Binance.com"
+
+    def __init__(
+        self,
+        account_holder: str,
+        api_key: str,
+        api_secret: str,
+        native_fiat: str,
+        username: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder, native_fiat)
+        self.__logger: logging.Logger = create_logger(f"{self.__BINANCE_COM}/{self.account_holder}")
+        self.__cache_key: str = f"{self.__BINANCE_COM.lower()}-{account_holder}"
+        self.client: binance = binance(

```

Comment:
Does this initiate the connection? If so it would be best to put it in the `load` method, otherwise it's fine here.

Reply:
`client` should have two leading underscores, since it's a private instance variable